File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.swift
import Foundation

func solution(_ word:String) -> Int {
    var data:[String] = []
    
    func find(_ depth: Int, _ current: String) {
        if depth == 6 { return }
        data.append(String(current))
        
        for i in ["A", "E", "I", "O", "U"] {
            find(depth+1, current + String(i))
        }
    }
    
    find(0, "")
    
    for i in 0..<data.count {
        if data[i] == word {
            return i
        }
    }
    
    return 0
}

File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = []
    
    def dfs(w, length):
        if length > 5:
            return
        answer.append(w)
        
        for i in ["A", "E", "I", "O", "U"]:
            dfs(w + i, length+1)
            
    dfs("", 0)        
        
    return answer.index(word)